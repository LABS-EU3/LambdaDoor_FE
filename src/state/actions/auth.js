/* eslint-disable import/prefer-default-export */
import axios from 'axios';
import jwt from 'jsonwebtoken';
import * as types from '../types';

export const setAuthenticated = (userId, name, email, profilePicture) => {
  return {
    type: types.LOG_IN_USER_SUCCESS,
    payload: {
      userId,
      name,
      email,
      profilePicture,
    },
  };
};

export const loginUser = (
  userId,
  name,
  email,
  profilePicture
) => async dispatch => {
  dispatch({
    type: types.LOG_IN_USER,
  });

  try {
    await axios.post('https://lambdadoor-staging.herokuapp.com/users', {
      slack_id: userId,
      name,
      email_address: email,
      img_72: profilePicture,
    });

    const token = jwt.sign(
      { userId, name, email, profilePicture },
      process.env.REACT_APP_JWT_SECRET
    );
    localStorage.setItem('token', token);

    dispatch(setAuthenticated(userId, name, email, profilePicture));
  } catch (error) {
    dispatch({
      type: types.LOG_IN_USER_FAILURE,
      payload: error.message,
    });
  }
};
