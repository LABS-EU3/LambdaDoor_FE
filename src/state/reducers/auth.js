import * as types from '../types';

const initialState = {
  isLoggingIn: false,
  isLoggedIn: false,
};

export const authState = (state = initialState, action) => {
  switch (action.type) {
    case types.LOG_IN_USER:
      return {
        ...state,
        isLoggingIn: true,
      };
    case types.GET_COMPANY_REVIEWS_SUCCESS:
      return {
        ...state,
        isLoggedIn: true,
      };
    case types.GET_COMPANY_REVIEWS_FAILURE:
      return state;
    default:
      return state;
  }
};
