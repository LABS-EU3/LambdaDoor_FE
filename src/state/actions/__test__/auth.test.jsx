import React from 'react';
import * as rtl from '@testing-library/react';
// import configureStore from 'redux-mock-store';
// import axios from 'axios';
// import MockAdapter from 'axios-mock-adapter';
// import thunk from 'redux-thunk';
// import * as types from '../../types/index';
import { SetAuthenticated, LoginUser, LogoutUser } from '../auth';

// const middlewares = [thunk];
// const mockStore = configureStore(middlewares);
// const mock = new MockAdapter(axios);
// const cred = {
//   withCredentials: true,
// };

// const backend = [
//   {
//     id: 1,
//     name: 'Bob Smith',
//     image: <img src="one.jpeg" alt="Bob's pic" />,
//   },
// ];

afterEach(rtl.cleanup);
let wrapper;
beforeEach(() => {
  wrapper = rtl.render;
});

describe('Action/types Auth testing', () => {
  //   it('should execute get review data data', async () => {
  //     const URL = `${backend}/${backend.id}`;
  //     mock.onGet(URL).reply(200, cred);
  //     const store = mockStore({});
  //     const actions = store.getActions();

  //     await store.dispatch(SetAuthenticated());
  //     expect(actions[0]).toEqual({ type: types.LOG_IN_USER });
  //   });

  it('Displays a snapshot for Auth testing', () => {
    const { asFragment } = wrapper(<SetAuthenticated />);
    expect(wrapper(<SetAuthenticated />).container).toMatchSnapshot();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Displays a snapshot for Logout testing', () => {
    const { asFragment } = wrapper(<LogoutUser />);
    expect(wrapper(<LogoutUser />).container).toMatchSnapshot();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Displays a snapshot for Login testing', () => {
    const { asFragment } = wrapper(<LoginUser />);
    expect(wrapper(<LoginUser />).container).toMatchSnapshot();
    expect(asFragment()).toMatchSnapshot();
  });
});
