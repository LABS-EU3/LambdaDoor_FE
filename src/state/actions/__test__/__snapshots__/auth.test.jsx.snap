// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action/types Auth testing Displays a snapshot for Auth testing 1`] = `<div />`;

exports[`Action/types Auth testing Displays a snapshot for Auth testing 2`] = `<DocumentFragment />`;

exports[`Action/types Auth testing Displays a snapshot for Login testing 1`] = `<div />`;

exports[`Action/types Auth testing Displays a snapshot for Login testing 2`] = `<DocumentFragment />`;

exports[`Action/types Auth testing Displays a snapshot for Logout testing 1`] = `<div />`;

exports[`Action/types Auth testing Displays a snapshot for Logout testing 2`] = `<DocumentFragment />`;
